analyzers/src/SonarAnalyzer.CSharp/Rules/MemberShadowsOuterStaticMember.cs

```diff
@@ -47,42 +47,34 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var innerClassSymbol = (INamedTypeSymbol)c.Symbol;
                     var containerClassSymbol = innerClassSymbol.ContainingType;
-                    if (!innerClassSymbol.IsClass() ||
-                        !containerClassSymbol.IsClass())
+                    if (!innerClassSymbol.IsClassOrStruct() ||
+                        !containerClassSymbol.IsClassOrStruct())
                     {
                         return;
                     }
 
                     var members = innerClassSymbol.GetMembers().Where(member => !member.IsImplicitlyDeclared);
                     foreach (var member in members)
                     {
-                        if (member is IPropertySymbol property)
+                        switch (member)
                         {
-                            CheckProperty(c, containerClassSymbol, property);
-                            continue;
-                        }
-
-                        if (member is IFieldSymbol field)
-                        {
-                            CheckField(c, containerClassSymbol, field);
-                            continue;
-                        }
-
-                        if (member is IEventSymbol @event)
-                        {
-                            CheckEvent(c, containerClassSymbol, @event);
-                            continue;
-                        }
-
-                        if (member is IMethodSymbol method)
-                        {
-                            CheckMethod(c, containerClassSymbol, method);
-                            continue;
-                        }
-
-                        if (member is INamedTypeSymbol namedType)
-                        {
-                            CheckNamedType(c, containerClassSymbol, namedType);
+                            case IPropertySymbol property:
+                                CheckProperty(c, containerClassSymbol, property);
+                                break;
+                            case IFieldSymbol field:
+                                CheckField(c, containerClassSymbol, field);
+                                break;
+                            case IEventSymbol @event:
+                                CheckEvent(c, containerClassSymbol, @event);
+                                break;
+                            case IMethodSymbol method:
+                                CheckMethod(c, containerClassSymbol, method);
+                                break;
+                            case INamedTypeSymbol namedType:
+                                CheckNamedType(c, containerClassSymbol, namedType);
+                                break;
+                            default:
```

Comment:
I'd be fine with not having the `default` here too. And ignoring the rule that will ask to have it. It's just a noise.